apps/backend/src/app/modules/maps/maps.service.spec.ts

```diff
@@ -206,4 +212,41 @@ describe('MapsService', () => {
       });
     }
   });
+
+  describe('submitMap', () => {
+    it('should throw ConflictException if map with the same bspHash already exists', async () => {
+      const dto = {} as CreateMapDto;
+      const userID = 1;
+      const bspHash = 'existingHash';
+      const bspFile = { buffer: Buffer.from('dummy data') } as any;
+
+      jest.spyOn(service, 'checkCreateDto').mockResolvedValueOnce(undefined);
+      jest.spyOn(service, 'getBspFromTemp').mockResolvedValueOnce(bspFile);
+      jest
+        .spyOn(service, 'checkMapCompression')
+        .mockResolvedValueOnce(undefined);
+      jest.spyOn(service, 'checkMapFiles').mockImplementationOnce(() => {});
+      jest
+        .spyOn(service, 'checkSuggestionsAndZones')
+        .mockImplementationOnce(() => {});
+      jest
+        .spyOn(FileStoreService, 'getHashForBuffer')
+        .mockReturnValueOnce(bspHash);
+
+      db.mMap.findFirst.mockResolvedValueOnce({
+        id: 123,
+        name: '123',
+        status: 123,
+        images: [],
+        submitterID: 123,
+        currentVersionID: '123',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      });
+
+      await expect(service.submitMap(dto, userID, [])).rejects.toThrow(
+        ConflictException
+      );
+    });
+  });
```

Comment:
This should be an E2E test. They're a bit more complicated to setup I'm afraid (and admittedly we E2E test too much), but all the related tests are done as E2E, should be easy to fit hash check stuff in there.